### PR TITLE
Improve the docs

### PR DESCRIPTION
**Pull request by ironcladlou**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 
-The Cluster DNS Operator manages cluster DNS services for [OpenShift](https://openshift.io/). 
+Cluster DNS Operator enables [DNS-based Kubernetes Service discovery](https://kubernetes.io/docs/concepts/services-networking/service/#dns) in [OpenShift](https://openshift.io) by managing [CoreDNS](https://coredns.io).
 
-## Development
+* The default cluster domain is `cluster.local`.
+* Configuration of the CoreDNS [Corefile](https://coredns.io/manual/toc/#configuration) or [kubernetes plugin](https://coredns.io/plugins/kubernetes/) are not yet supported.
 
-See [HACKING](HACKING.md) for development topics.
+The operator tries to be useful by default.
```
[the remark this answers is not in the document]
Comment (ironcladlou) [reply README.md]: Tried again, what do you think?

File: README.md [hunk 2]
```diff
@@ -1,7 +1,16 @@
 # Cluster DNS Operator
 
-The Cluster DNS Operator manages cluster DNS services for [OpenShift](https://openshift.io/). 
+Cluster DNS Operator enables [DNS-based Kubernetes Service discovery](https://kubernetes.io/docs/concepts/services-networking/service/#dns) in [OpenShift](https://openshift.io) by managing [CoreDNS](https://coredns.io).
```
[the remark this answers is not in the document]
Comment (ironcladlou) [reply README.md]: Tried again, what do you think?

**Comment by Miciah** [comment]
The providence of the name resolution service's address, and the fact that it is hard-coded to a .10 address, is under- or undocumented and was a source of confusion for us earlier on.  Also, the statement that the default configuration is "based on the cluster's configuration" is left unsubstantiated: As far as what is in the default configuration, the document only mentions cluster domain, which gets a hard-coded value, so the reader may wonder what the cluster configuration is actually used for.  It might be worth adding another bullet-point to document where the address comes from and to substantiate the "cluster's configuration" statement: "The name resolution service's address is `x.y.z.10` where `x.y.z` comes from the cluster's service CIDR."

**Comment by Miciah** [comment]
We discussed and decided that documenting the name resolution service's address and mentioning the cluster's service CIDR is superfluous for the README.
/lgtm

**Comment by ironcladlou** [comment]
I don't know yet whether the IP allocation algorithm is important enough to describe here — my main concern is that it deserves cross-referencing at least one orthogonal concept (the service network CIDR from the cluster API config which doesn't even yet exist). It would be nice to start trying to figure out how to make these decisions about what belongs in the README vs. the product docs...

**Pull request merged**